src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -339,6 +339,52 @@ instruct storeV_partial(vReg src, vmemA mem, pRegGov pTmp, iRegINoSp tmp, rFlags
   ins_pipe(pipe_slow);
 %}
 
+// vector to mask
+
+instruct vector2vmask(pRegGov pg, vReg src, rFlagsReg cr) %{
+  predicate(UseSVE > 0);
+  match(Set pg (VectorToMask src));
+  effect(KILL cr, DEF pg, USE src);
+  ins_cost(SVE_COST);
+  format %{ "sve_cmpeq  $pg, $src, -1\t# vector mask (sve)" %}
+  ins_encode %{
+    BasicType bt = vector_element_basic_type(this, $src);
+    __ sve_cmpeq(as_PRegister($pg$$reg), elemType_to_regVariant(bt),
+                 ptrue, as_FloatRegister($src$$reg), -1);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcmp2vmask(pRegGov pg, vReg src1, vReg src2, immI cond, rFlagsReg cr)  %{
+  predicate(UseSVE > 0);
+  match(Set pg (VectorToMask (VectorMaskCmp (Binary src1 src2) cond)));
+  effect(KILL cr, DEF pg);
+  ins_cost(SVE_COST);
+  format %{ "sve_cmp $pg, $src1, $src2\t# vector cmp (sve)" %}
+  ins_encode %{
+    BasicType bt = this->in(1)->bottom_type()->is_vect()->element_basic_type();
+    sve_compare(C2_MacroAssembler(&cbuf), as_PRegister($pg$$reg), bt,
+                ptrue, as_FloatRegister($src1$$reg),
+                as_FloatRegister($src2$$reg), (int)$cond$$constant);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// mask to vector
+
+instruct vmask2vector(vReg dst, pRegGov pg) %{
+  predicate(UseSVE > 0);
+  match(Set dst (MaskToVector pg));
```

Comment:
We already have VectorLoadMask and VectorStoreMask nodes which have same semantics as Mask2Vector and Vector2Mask operation. Can you please elaborate the need for these new nodes.

Reply:
Thanks for your comment about this. `VectorLoadMask` and `VectorStoreMask` are vector nodes that have the `TypeVect` type, while `Vector2Mask` is a mask node that have the `TypeVMask` type.  For SVE, the two new added nodes are used for conversions between vector register and predicate register, e.g. `Mask2Vector` for `predicate -> vector` and `Vector2Mask` for `vector -> predicate`. Yes, it seems `VectorLoadMask` and `VectorStoreMask` can match the same requirements like what have been done in the AVX-512 patch. However, in our whole proposal, we defined different IRs to separate vector and mask operations. I think the main conflict is the IR part for mask. And it's better to get a final solution to the different operations together for SVE and AVX-512.

Reply:
Note that `MaskToVector` can be used for API `VectorMask.toVector()` in future which is implemented with `blend` currently.

Reply:
> Thanks for your comment about this. `VectorLoadMask` and `VectorStoreMask` are vector nodes that have the `TypeVect` type, while `Vector2Mask` is a mask node that have the `TypeVMask` type. For SVE, the two new added nodes are used for conversions between vector register and predicate register, e.g. `Mask2Vector` for `predicate -> vector` and `Vector2Mask` for `vector -> predicate`. Yes, it seems `VectorLoadMask` and `VectorStoreMask` can match the same requirements like what have been done in the AVX-512 patch. However, in our whole proposal, we defined different IRs to separate vector and mask operations. I think the main conflict is the IR part for mask. And it's better to get a final solution to the different operations together for SVE and AVX-512.

Hi @XiaohongGong , 
Yes currently all the ideal nodes have fixed types , but technically there is nothing incorrect in having nodes (VectorLoadMask/VectorStoreMask) with types based on the target features, e.g. VectorLoadMask may have TypeVect for targets which do-not support predicate registers and a predicate vector type (TypeVectMask/TypeVMask) otherwise. 
By taking this approach we may be able to save this synthetic duplication for masked flavors of vector operation nodes.